src/lib.rs

```diff
@@ -250,64 +250,39 @@ impl BootInformationInner {
 
 impl fmt::Debug for BootInformation {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        writeln!(f, "multiboot information")?;
-
-        writeln!(
-            f,
-            "S: {:#010X}, E: {:#010X}, L: {:#010X}",
-            self.start_address(),
-            self.end_address(),
-            self.total_size()
-        )?;
-
-        if let Some(boot_loader_name_tag) = self.boot_loader_name_tag() {
-            writeln!(f, "boot loader name: {}", boot_loader_name_tag.name())?;
-        }
-
-        if let Some(command_line_tag) = self.command_line_tag() {
-            writeln!(f, "command line: {}", command_line_tag.command_line())?;
-        }
+        let mut module_tags_count = 0;
+        let mut kernel_section_count = 0;
 
-        if let Some(memory_map_tag) = self.memory_map_tag() {
-            writeln!(f, "memory areas:")?;
-            for area in memory_map_tag.memory_areas() {
-                writeln!(
-                    f,
-                    "    S: {:#010X}, E: {:#010X}, L: {:#010X}",
-                    area.start_address(),
-                    area.end_address(),
-                    area.size()
-                )?;
-            }
+        for _t in  self.module_tags() {
```

Comment:
`module_tags` returns an ~`impl Iterator` so this could just be `self.module_tags().count` I think